ams/ams-server/src/main/java/com/netease/arctic/ams/server/service/impl/TableExpireService.java

```diff
@@ -103,69 +101,94 @@ public void run() {
-            if (changeTable == null) {
-              LOG.warn("[{}] Change table is null: {}", traceId, tableIdentifier);
-              return null;
-            }
+  public static void expireArcticTable(ArcticTable arcticTable) {
+    TableIdentifier tableIdentifier = arcticTable.id();
+    final String traceId = UUID.randomUUID().toString();
+    long startTime = System.currentTimeMillis();
+    LOG.info("[{}] {} start expire", traceId, tableIdentifier);
+
+    long changeDataTTL = Long.parseLong(arcticTable.properties()
+        .getOrDefault(TableProperties.CHANGE_DATA_TTL,
+            TableProperties.CHANGE_DATA_TTL_DEFAULT)) * 60 * 1000;
+    long baseSnapshotsKeepTime = Long.parseLong(arcticTable.properties()
+        .getOrDefault(TableProperties.BASE_SNAPSHOT_KEEP_MINUTES,
+            TableProperties.BASE_SNAPSHOT_KEEP_MINUTES_DEFAULT)) * 60 * 1000;
+    long changeSnapshotsKeepTime = Long.parseLong(arcticTable.properties()
+        .getOrDefault(TableProperties.CHANGE_SNAPSHOT_KEEP_MINUTES,
+            TableProperties.CHANGE_SNAPSHOT_KEEP_MINUTES_DEFAULT)) * 60 * 1000;
 
-            // get valid files in the change store which shouldn't physically delete when expire the snapshot
-            // in the base store
-            Set<String> baseExclude = UnKeyedTableUtil.getAllContentFilePath(changeTable);
-            baseExclude.addAll(finalHiveLocation);
-            expireSnapshots(baseTable, startTime - baseSnapshotsKeepTime, baseExclude);
-            long baseCleanedTime = System.currentTimeMillis();
-            LOG.info("[{}] {} base expire cost {} ms", traceId, arcticTable.id(), baseCleanedTime - startTime);
-
-            // delete ttl files
-            List<DataFileInfo> changeDataFiles = ServiceContainer.getFileInfoCacheService()
-                .getChangeTableTTLDataFiles(keyedArcticTable.id().buildTableIdentifier(),
-                    System.currentTimeMillis() - changeDataTTL);
-            deleteChangeFile(keyedArcticTable, changeDataFiles);
-
-            // get valid files in the base store which shouldn't physically delete when expire the snapshot
-            // in the change store
-            Set<String> changeExclude = UnKeyedTableUtil.getAllContentFilePath(baseTable);
-            changeExclude.addAll(finalHiveLocation);
-            expireSnapshots(changeTable, startTime - changeSnapshotsKeepTime, changeExclude);
-            return null;
-          });
-          LOG.info("[{}] {} expire cost total {} ms", traceId, arcticTable.id(),
-              System.currentTimeMillis() - startTime);
-        } else {
-          UnkeyedTable unKeyedArcticTable = arcticTable.asUnkeyedTable();
-          expireSnapshots(unKeyedArcticTable, startTime - baseSnapshotsKeepTime, hiveLocation);
-          long baseCleanedTime = System.currentTimeMillis();
-          LOG.info("[{}] {} unKeyedTable expire cost {} ms", traceId, arcticTable.id(), baseCleanedTime - startTime);
+    Set<String> hiveLocation = new HashSet<>();
```

Comment:
Change parameter name to `hiveLocations` or `hiveLocationSet`?

Reply:
I changed it to `hiveLocations`